src/glyphnote.ts

```diff
@@ -36,31 +41,54 @@ export class GlyphNote extends Note {
     return this.glyph.getBoundingBox();
   }
 
-  /*
-  addToModifierContext() {
+  // Add self to modifier context. `mContext` is the `ModifierContext`
+  // to be added to.
+  addToModifierContext(mContext: ModifierContext): this {
+    this.modifierContext = mContext;
+    for (let i = 0; i < this.modifiers.length; ++i) {
+      this.modifierContext.addMember(this.modifiers[i]);
+    }
+    this.setPreFormatted(false);
     return this;
   }
-  */
 
   preFormat(): this {
+    if (!this.preFormatted && this.modifierContext) {
+      this.modifierContext.preFormat();
+    }
     this.setPreFormatted(true);
     return this;
   }
-
+  drawModifiers(): void {
+    const ctx = this.checkContext();
+    ctx.openGroup('modifiers');
+    for (let i = 0; i < this.modifiers.length; i++) {
+      const modifier = this.modifiers[i];
+      modifier.setContext(ctx);
+      modifier.drawWithStyle();
+    }
+    ctx.closeGroup();
+  }
   draw(): void {
-    const stave = this.checkStave();
-    const ctx = stave.checkContext();
+    if (!this.stave) {
+      throw new Vex.RERR('NoStave', 'No stave attached to this note.');
+    }
+
+    const ctx = this.stave.checkContext();
```

Comment:
`this.checkStave()` works with `this.stave` like `this.checkContext()` with `this.context`:
- `this.checkStave()` will throw an `RERR` if `this.stave` is `undefined`. 

Please revert the change.

Reply:
Sorry, that must have snuck in from my other branch.  I reverted to what it was.

Reply:
This is not reverted, did you push the change?